app/shared/components/year-with-line/YearWithLine.tsx

```diff
@@ -0,0 +1,52 @@
+'use client';
+import { Box } from '@mui/system';
+import { useEffect, useRef, useState } from 'react';
+
+import { oswald, theme } from '~/ds-components/theme/Theme';
+import useBreakpoints from '~/hooks/use-breakpoints/useBreakpoints';
+
+import { styles } from './YearWithLine.styles';
+const YearWithLine = ({ year }: { year: number }) => {
+  const Styles = styles(theme);
+
+  const numberStr = year.toString();
+  const [offset, setOffset] = useState(0);
+  const textRef = useRef<HTMLDivElement>(null);
+  const { isTablet, isMobile } = useBreakpoints();
+
+  useEffect(() => {
+    const calculateOffset = () => {
+      if (!textRef.current) return;
+
+      const canvas = document.createElement('canvas');
+      const ctx = canvas.getContext('2d');
+      if (!ctx) return;
+
+      let fontSize = 236;
+      if (isTablet) fontSize = 132;
+      if (isMobile) fontSize = 114;
+      ctx.font = `500 ${fontSize}px ${oswald.style.fontFamily}`;
```

Comment:
Maybe it would be better to use these values from the theme instead of hardcoding them? This suggestion might look a bit tricky, but this keeps styles in sync if the theme changes
```suggestion
      let fontSize = parseInt(theme.typography.customBold236.fontSize as string, 10);
      if (isTablet) fontSize = parseInt(theme.typography.customBold132.fontSize as string, 10);
      if (isMobile) fontSize = parseInt(theme.typography.customBold114.fontSize as string, 10);

      ctx.font = `${theme.typography.customBold236.fontWeight} ${fontSize}px ${theme.typography.customBold236.fontFamily}`;
```

Reply:
Good idea, I agree with it
But I wont change pixels to numbers yet so as not to mess up the code-style for typography